test/e2e/main_test.go

```diff
@@ -449,7 +449,8 @@ func TestExternalService(t *testing.T) {
 
 func exposeServiceSpec() corev1.ServiceSpec {
 	return corev1.ServiceSpec{
-		Type: exposeServiceType(),
+		Type:  exposeServiceType(),
+		Ports: []corev1.ServicePort{{NodePort: 30222}},
```

Comment:
Having NodePort hard-coded in the tests is what could cause issues on particular platforms. Wouldn't it be better to retrieve actual port used after service creation?

Reply:
I have to check how kind works in CI, it seems it need 30222

Reply:
yep it's configured at cluster startup. I would keep this and provide more configurability in a separated PR if needed